src/main/scala/com/scylladb/migrator/readers/Cassandra.scala

```diff
@@ -126,7 +128,17 @@ object Cassandra {
 
       timestampsToFields
         .map {
-          case ((ttl, writetime), fields) =>
+          case ((ttl, writetime), fields) => {
+            val writetimestamp = writetime.getOrElse(CassandraOption.Unset)
+            val baseTTL = if (defaultTTL > 0) defaultTTL else 0L
+            val deltaTTL = if (writetimestamp == CassandraOption.Unset) {
+              0
+            } else {
+              baseTTL - (System.currentTimeMillis - writetimestamp.asInstanceOf[Long] / 1000) / 1000
+            }
+            // expire the record in 1s in case delta is negative, use given TTL if write timestamp wasn't found
+            val finalttl =
+              if (deltaTTL > 0) deltaTTL else if (deltaTTL == 0) baseTTL else 1L
```

Comment:
Here use assume that `deltaTTL == 0` is always the case when the write timestamp wasn’t found. But it seems it could also happen e.g. if `baseTTL == 5` and the write timestamp was 5 seconds ago.

Maybe use the type `Option` again to model the fact that the `deltaTTL` might be unset. So, you would have something like:

```suggestion
              deltaTTL.map(_.max(1)).getOrElse(baseTTL)
```